### PR TITLE
Updated grpc ebook for NET 6

### PR DESCRIPTION
**Pull request by VaishaliRaval**
Updated grpc ebook for NET 6

### CODE REVIEW

File: docs/architecture/grpc-for-wcf-developers/create-project.md [hunk 1]
```diff
@@ -93,28 +93,45 @@ In the gRPC build targets, there's a `Protobuf` item element that lets you speci
...
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+app.MapGrpcService<PortfolioService>();
+app.MapGet("/", () => "Communication with gRPC endpoints must be made through a gRPC client. To learn how to create a client, visit: https://go.microsoft.com/fwlink/?linkid=2086909");
+
+app.Run();
+```
+
+Using a `GlobalUsing` file, which applies to all the source file in the project and includes all the namespaces makes it easy to maintain them at one place
```
Comment (nishanil) [inline docs/architecture/grpc-for-wcf-developers/create-project.md:121]: For this sample, we don't require a `GlobalUsing` file. Can you drop this one, please?

File: docs/architecture/grpc-for-wcf-developers/call-credentials.md [hunk 2]
```diff
@@ -18,36 +18,36 @@ For more information on how to get started with this authentication method, see
...
+ASP.NET Core 6.0 can handle JWTs by using the JWT Bearer package. The configuration is exactly the same for a gRPC application as it is for an ASP.NET Core MVC application. Here, we'll focus on JWT Bearer tokens, because they're easier to develop with than WS-Federation.
 
 ## Add authentication and authorization to the server
 
-The JWT Bearer package isn't included in ASP.NET Core 5.0 by default. Install the [Microsoft.AspNetCore.Authentication.JwtBearer](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.JwtBearer) NuGet package in your app.
+The JWT Bearer package isn't included in ASP.NET Core 6.0 by default. Install the [Microsoft.AspNetCore.Authentication.JwtBearer](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.JwtBearer) NuGet package in your app.
 
-Add the Authentication service in the Startup class, and configure the JWT Bearer handler:
+Add the Authentication service in the Program.cs class, and configure the JWT Bearer handler:
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/call-credentials.md:27]:
```suggestion
Add the Authentication service in the _Program.cs_ class, and configure the JWT Bearer handler:
```

File: docs/architecture/grpc-for-wcf-developers/call-credentials.md [hunk 3]
```diff
@@ -69,16 +69,10 @@ Next, add the Authorization service, which controls access to the system:
 > [!TIP]
 > Authentication and authorization are two separate steps. You use authentication to determine the user's identity. You use authorization to decide whether that user is allowed to access various parts of the system.
 
-Now add the authentication and authorization middleware to the ASP.NET Core pipeline in the `Configure` method:
+Now add the authentication and authorization middleware to the ASP.NET Core pipeline in the `Program.cs`:
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/call-credentials.md:72]:
```suggestion
Now add the authentication and authorization middleware to the ASP.NET Core pipeline in the _Program.cs_:
```

When referencing a file, the text should be italic.

File: docs/architecture/grpc-for-wcf-developers/channel-credentials.md [hunk 4]
```diff
@@ -20,27 +20,22 @@ Configure certificate authentication both at the host level (for example, on the
 
 ### Configure certificate validation on Kestrel
 
-You can configure Kestrel (the ASP.NET Core HTTP server) to require a client certificate, and optionally to carry out some validation of the supplied certificate, before accepting incoming connections. You specify this configuration in the `CreateWebHostBuilder` method of the `Program` class, rather than in `Startup`.
+You can configure Kestrel (the ASP.NET Core HTTP server) to require a client certificate, and optionally to carry out some validation of the supplied certificate, before accepting incoming connections. You specify this configuration in the `Program.cs`
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/channel-credentials.md:23]:
```suggestion
You can configure Kestrel (the ASP.NET Core HTTP server) to require a client certificate, and optionally to carry out some validation of the supplied certificate, before accepting incoming connections. You specify this configuration in the _Program.cs_:
```

File: docs/architecture/grpc-for-wcf-developers/channel-credentials.md [hunk 5]
```diff
@@ -49,43 +44,33 @@ The `ClientCertificateMode.RequireCertificate` setting causes Kestrel to immedia
 
 The [Microsoft.AspNetCore.Authentication.Certificate](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Certificate) NuGet package provides certificate authentication.
 
-Add the certificate authentication service in the `ConfigureServices` method, and add authentication and authorization to the ASP.NET Core pipeline in the `Configure` method.
+Add the certificate authentication service in the `Program.cs`, and add authentication and authorization to the ASP.NET Core pipeline.
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/channel-credentials.md:47]:
```suggestion
Add the certificate authentication service in the _Program.cs_, and add authentication and authorization to the ASP.NET Core pipeline.
```

File: docs/architecture/grpc-for-wcf-developers/client-libraries.md [hunk 6]
```diff
@@ -28,30 +28,25 @@ The `IObservable<T>` interface is the "reactive" inverse of `IEnumerable<T>`. Ra
 This code is longer than the `IAsyncEnumerable<T>` code, because C# doesn't have built-in support for working with observables. You have to create the implementation class manually. It's a generic class, though, so a single implementation works across all types.
 
 ```csharp
-using System;
-using System.Threading;
-
-namespace Grpc.Core
+namespace Grpc.Core;
+public class GrpcStreamObservable<T> : IObservable<T>
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/client-libraries.md:32]:
```suggestion

public class GrpcStreamObservable<T> : IObservable<T>
```

File: docs/architecture/grpc-for-wcf-developers/create-project.md [hunk 7]
```diff
@@ -93,28 +93,29 @@ In the gRPC build targets, there's a `Protobuf` item element that lets you speci
...
 
 ```csharp
-public class PortfolioService : Portfolios.PortfoliosBase
+public class PortfolioService : Protos.Portfolios.PortfoliosBase
 {
 }
 ```
 
-There was a reference to the `GreeterService` class in the `Configure` method in the `Startup` class. If you used refactoring to rename the class, this reference should have been updated automatically. However, if you didn't, you need to edit it manually.
+There was a reference to the `GreeterService` class in the `Program.cs`. If you used refactoring to rename the class, this reference should have been updated automatically. However, if you didn't, you need to edit it manually.
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/create-project.md:101]:
```suggestion
There was a reference to the `GreeterService` class in the _Program.cs_. If you used refactoring to rename the class, this reference should have been updated automatically. However, if you didn't, you need to edit it manually.
```

File: docs/architecture/grpc-for-wcf-developers/create-project.md (hunk 1 above)
Comment (VaishaliRaval) [reply docs/architecture/grpc-for-wcf-developers/create-project.md]: Yes sure, I have dropped GlobalUsing

File: docs/architecture/grpc-for-wcf-developers/migrate-request-reply.md [hunk 8]
```diff
@@ -177,11 +177,9 @@ If you save your project with these changes, the gRPC build target will run in t
 Open the `Services/GreeterService.cs` class and delete the example code. Now you can add the Portfolio service implementation. The generated base class will be in the `Protos` namespace and is generated as a nested class. gRPC creates a static class with the same name as the service in the `.proto` file and a base class with the suffix `Base` inside that static class, so the full identifier for the base type is `TraderSys.Portfolios.Protos.Portfolios.PortfoliosBase`.
 
 ```csharp
-namespace TraderSys.Portfolios.Services
+namespace TraderSys.Portfolios.Services;
+public class PortfolioService : Protos.Portfolios.PortfoliosBase
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/migrate-request-reply.md:181]:
```suggestion

public class PortfolioService : Protos.Portfolios.PortfoliosBase
```

File: docs/architecture/grpc-for-wcf-developers/migrate-request-reply.md [hunk 9]
```diff
@@ -231,21 +229,25 @@ public class PortfolioItem
 
 ## Use ASP.NET Core dependency injection
 
-Now you can add a reference to this library to the gRPC application project and consume the `PortfolioRepository` class by using dependency injection in the gRPC service implementation. In the WCF application, dependency injection was provided by the Autofac IoC container. ASP.NET Core has dependency injection baked in. You can register the repository in the `ConfigureServices` method in the `Startup` class:
+Now you can add a reference to this library to the gRPC application project and consume the `PortfolioRepository` class by using dependency injection in the gRPC service implementation. In the WCF application, dependency injection was provided by the Autofac IoC container. ASP.NET Core has dependency injection baked in. You can register the repository in the `Program.cs` itself:
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/migrate-request-reply.md:232]:
```suggestion
Now you can add a reference to this library to the gRPC application project and consume the `PortfolioRepository` class by using dependency injection in the gRPC service implementation. In the WCF application, dependency injection was provided by the Autofac IoC container. ASP.NET Core has dependency injection baked in. You can register the repository in the _Program.cs_ itself:
```

File: docs/architecture/grpc-for-wcf-developers/migrate-request-reply.md [hunk 10]
```diff
@@ -303,40 +305,38 @@ After there's a proper `Guid` value for `traderId`, you can use the repository t
 The previous code doesn't actually work because the repository is returning its own POCO model `Portfolio`, but gRPC needs its own Protobuf message `Portfolio`. As when you map Entity Framework types to data transfer types, the best solution is to provide a conversion between the two. A good place to put the code for this conversion is in the Protobuf-generated class, which is declared as a `partial` class so it can be extended:
 
 ```csharp
-namespace TraderSys.Portfolios.Protos
+namespace TraderSys.Portfolios.Protos;
+public partial class PortfolioItem
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/migrate-request-reply.md:309]:
```suggestion

public partial class PortfolioItem
```

File: docs/architecture/grpc-for-wcf-developers/protobuf-messages.md [hunk 11]
```diff
@@ -15,20 +15,18 @@ The rest of the chapter will look in more detail at how different types of data
 In Windows Communication Foundation (WCF), a `Stock` class for a stock market trading application might be defined like the following example:
 
 ```csharp
-namespace TraderSys
+namespace TraderSys;
+[DataContract]
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/protobuf-messages.md:19]:
```suggestion

[DataContract]
```

File: docs/architecture/grpc-for-wcf-developers/wcf-services-to-grpc-comparison.md [hunk 12]
```diff
@@ -44,27 +44,21 @@ message HelloReply {
 ```
 
 ```csharp
-using System.Threading.Tasks;
-using Grpc.Core;
-using Microsoft.Extensions.Logging;
-
-namespace HelloGrpc
+namespace HelloGrpc;
+public class GreeterService : Greeter.GreeterBase
```
Comment (IEvangelist) [inline docs/architecture/grpc-for-wcf-developers/wcf-services-to-grpc-comparison.md:48]:
```suggestion

public class GreeterService : Greeter.GreeterBase
```

**Review by IEvangelist — CHANGES_REQUESTED** [review]
Looks great, just a few suggestions.

File: docs/architecture/grpc-for-wcf-developers/call-credentials.md (hunk 3 above)
Comment (VaishaliRaval) [reply docs/architecture/grpc-for-wcf-developers/call-credentials.md]: Ok sure

**Review by nishanil — APPROVED** [review]
Looks good. We will merge this.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

**Pull request merged**